test/test_autograd.py

```diff
@@ -4080,6 +4079,16 @@ def test_gradcheck_check_analytical_jacobian_attributes(self):
         # TODO: when grad_input is incorrect dtype/size
         pass
 
+    def test_gradcheck_jacobian_mismatch(self):
+        def fn(x):
+            y = x.clone()
+            y.register_hook(lambda x: x + 1e-2)
+            return y
+        x = torch.ones(2, 2, requires_grad=True)
+        with self.assertRaisesRegex(RuntimeError, 'Jacobian mismatch for output 0 with respect to input 0'):
+            gradcheck(fn, (x,))
+        self.assertFalse(gradcheck(fn, (x,), raise_exception=False))
+
```

Comment:
There are a few gradcheck jacobian mismatch paths (this test is testing R->R functions), we should write tests to exercise all of them for completeness. It looks like checking a C->C function, C->R function, and R->C function would do the job